### PR TITLE
Adds "Feature" layout for visual storytelling (no g-modal component)

### PR DESCRIPTION
**Pull request by aendrew**
I got fed up with trying to make an a11y modal window and realised I don't actually need one so have closed both #119 and #123, cherrypicking the relevant commits off the latter to not include any modal trash.

### CODE REVIEW

File: components/feature/index.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import Header from '../header';
+import Analytics from '../analytics';
+import './styles.scss';
+import { useAds, useLayoutChangeEvents, useKeyboardShortcuts } from '../../shared/hooks';
+import { Context } from '../layout';
+import Progress from '../progress';
+
+export { Context } from '../layout';
+
+const Feature = forwardRef(
```
Comment (tlfrd) [inline components/feature/index.js]: Why is this using forwardRef? Is there something you're doing in your starter kit project that needs this?

Comment (aendrew) [reply components/feature/index.js]: This lets me pass a ref from the parent and then size things based on that.

File: components/feature/index.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+          {flags.header && <Header key="header" {...{ ...props, flags, breakpoint }} />}
+          <div ref={ref} className="g-feature__container">
+            <Progress
+              value={currentPage / Children.count(children)}
+              steps={Children.count(children) - 1}
+            />
+            {/* Navigation */}
+            {navigationEnabled && (
+              <nav className="g-feature__controls">
+                <div
```
Comment (joannaskao) [inline components/feature/index.js]: Should we make this a button instead of a div with a button role?

File: components/progress/styles.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  padding: 0;
+  margin: 0 0 30px 0;
+  display: flex;
+
+  &__segment {
+    margin-right: 2px;
+    flex: 1;
+    background: white;
+    &--active {
+      background: #ca1b54;
```
Comment (joannaskao) [inline components/progress/styles.scss]: I wonder if we should make the colour of the progress bar configurable. I can imagine we might want to use another colour depending on the background of the app.

File: components/feature/index.js (hunk 2 above)
Comment (aendrew) [reply components/feature/index.js]: I could but I'd have to remove all the styling and it wouldn't be much more semantic IMO. What's the benefit of doing that being semantic clarity?

File: components/feature/index.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
...o detraxit cu qui. Et elitr // dolore praesent est. // Ex his oratio perpetua expetendis, commune suscipiantur nec at, vel lorem atqui option ne. // Habeo congue vel te. Ex quo summo principes. Eu dicta solet molestiae eam, eu ius primis // consectetuer. Ei dicant numquam omittam qui. Vim dicunt nostrud volutpat ex, cu mel // gubergren constituto. // Ad qui wisi minimum cotidieque, ea vis esse discere intellegam. Et omnium inimicus vim. Ea // sale dicam homero ius. Impedit eligendi delicata in mel, vis soluta temporibus an, ex vis // vide laoreet. No per hinc ferri mediocritatem, eam modus ubique atomorum ne. Iudico offendit // ut his. // Mel fastidii eleifend no, ut oratio intellegam cum. Iudico dignissim sadipscing ei mel. At // laudem legendos erroribus eos. Aliquid dolorem dissentiunt eos cu. Eros suscipiantur mea ei, // tantas vituperata vis ad. Pri et dicunt regione, ut quando volutpat abhorreant eos. // Ad cum saperet docendi, ex habeo nemore euismod vix, usu semper instructior cu. Natum // salutatus consetetur sit cu. Ad erroribus complectitur vis. Ne possim similique vis. Ne enim // soluta vix, quis natum ne qui, ei vidit latine partiendo mei. Mel ea duis essent vivendo.`;
```
[the remark this answers is not in the document]
Comment (aendrew) [reply components/feature/index.stories.mdx]: Agreed. :+1:

File: components/progress/styles.scss (hunk 3 above)
Comment (aendrew) [reply components/progress/styles.scss]: Yep, totally 👍

Comment (aendrew) [reply components/progress/styles.scss]: Actually, have changed my mind about this. It's probably better to override this on the theme level using CSS for specific projects.

**Pull request merged**